modules/system/tests/bootstrap/TestCase.php

```diff
@@ -91,4 +92,22 @@ public static function assertRegExp(string $pattern, string $string, string $mes
 
         Assert::assertRegExp($pattern, $string, $message);
     }
+
+    /**
+     * Stub for `assertObjectHasAttribute` to allow compatibility with both PHPUnit 9 and 10.
+     *
+     * @param string $propertyName
+     * @param object $object
+     * @param string $message
+     * @return void
+     */
+    public static function assertObjectHasAttribute(string $propertyName, $object, string $message = ''): void
```

Comment:
Won't this fail to work as a drop in replacement as it is now a static method?

Reply:
Interestingly it was always a static method, just like assertRegExp, `$this->assertObjectHasProperty` calls the static method. I copied the stub from there, and it seems to work. Kinda strange indeed.

Reply:
@LukeTowers I think PHPUnit has a magic method that routes calls to the static methods.